### PR TITLE
feat: improve CI checks

### PR DESCRIPTION
**Pull request by Diiaablo95**
I keep catching Clippy warnings whenever cargo is run with a combination of features that is neither `--all-features` nor `default-features`, for instance with only `--features runtime-benchmarks`. So I thought having multiple checks in place might help to keep the code cleaner, and as all the checks are within the same CI step, it should not take THAT much longer for the checks to finish. 

If you think this is not a good idea, we can close it, but I though it would at least be worth discussing.

For reference, the [cargo check](https://doc.rust-lang.org/cargo/commands/cargo-check.html) and [cargo test](https://doc.rust-lang.org/cargo/commands/cargo-test.html) docs.

### CODE REVIEW

File: pallets/delegation/src/benchmarking.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ use frame_support::{
 use frame_system::RawOrigin;
 use sp_core::{offchain::KeyTypeId, sr25519};
 use sp_io::crypto::sr25519_generate;
-use sp_runtime::MultiSignature;
+use sp_runtime::{traits::Zero, MultiSignature};
```
Comment (weichweich) [inline pallets/delegation/src/benchmarking.rs:29]: why was this added?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,36 +1,36 @@
 stages:
+  - check
```
Comment (weichweich) [inline .gitlab-ci.yml:2]: I would clearly vote against that. This will slow down the CI.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
   timeout: 1 hours
   script:
     - rustup component add clippy --toolchain nightly
-    - cargo +nightly clippy --all-features --all-targets -- -D warnings
+    - cargo +nightly clippy --workspace --all-targets --no-default-features -- -D warnings
+    - cargo +nightly clippy --workspace --all-targets --features try-runtime -- -D warnings
+    - cargo +nightly clippy --workspace --all-targets --features runtime-benchmarks -- -D warnings
+    - cargo +nightly clippy --workspace --all-targets --features mock -- -D warnings
+    - cargo +nightly clippy --workspace --all-targets --features std -- -D warnings
+    - cargo +nightly clippy --workspace --all-targets --all-features -- -D warnings
```
Comment (weichweich) [inline .gitlab-ci.yml:17]: So we mostly miss unused import warnings. And the imports are only unused in very specific feature combinations? The disadvantages of a slow CI clearly outweigh the gains from me here. 😕

File: pallets/delegation/src/benchmarking.rs (hunk 1 above)
Comment (Diiaablo95) [reply pallets/delegation/src/benchmarking.rs:29]: This was a warning I caught, but it has been also addressed in https://github.com/KILTprotocol/mashnet-node/pull/265.

**Comment by Diiaablo95** [comment]
Alright, as it is not useful in any way I will close this.
